iceoryx2-bb/derive-macros/src/lib.rs

```diff
@@ -103,3 +103,83 @@ pub fn placement_default_derive(input: TokenStream) -> TokenStream {
 
     TokenStream::from(expanded)
 }
+
+/// Implements the [`iceoryx2_bb_elementary::zero_copy_send::ZeroCopySend`] trait when all fields of the struct implement it.
```

Comment:
There are two distinct and completely different proc macro implementations in one file. I think it would make sense to move them into separate files:
 * `placement_default.rs`
 * `zero_copy_send.rs`

Also the do not compile tests could be moved into a sub-directory so `tests/zero_copy_send_compile_tests.rs`

Reply:
Unfortunately, we cannot split the macro implementations. The compiler states that "functions tagged with #[proc_macro_derive] must currently reside in the root of the crate".

The compile tests are doctests and if I understood correctly, these cannot be in a separate test folder.

Reply:
You can get some inspiration for the structuring from here -> https://github.com/JelteF/derive_more/blob/master/impl/src/lib.rs

They also have compile tests, but at a quick glance, I could not find out how they are doing it.

Reply:
They use [trybuild](https://docs.rs/trybuild/latest/trybuild/) for the compile tests which allows to check if a test would fail to compile and to check the resulting error message. If we decide to use this lib, I would do this in a separate issue and PR where we also adapt the other compile tests of iceoryx2.